examples/ex33.hpp

```diff
@@ -0,0 +1,318 @@
+      Am.SetSize(h_Am,w_Am);
+      for (int i = 0; i<h_Am; i++)
+      {
+         int ii = J[i];
+         for (int j = 0; j<w_Am; j++)
+         {
+            Am(i,j) = A(ii,j);
+         }
+      }
+
+#ifdef MFEM_USE_LAPACK
+      DenseMatrixSVD svd(Am,false,true);
+      svd.Eval(Am);
+      DenseMatrix &v = svd.RightSingularvectors();
+      v.GetRow(k,w);
+#else
+      mfem_error("Compiled without LAPACK");
+#endif
+
+      // N = C*(w.*f); D = C*w; % numerator and denominator
+      Vector aux(w);
+      aux *= f_vec;
+      Vector N(C.Height()); // Numerator
+      C.Mult(aux,N);
+      Vector D(C.Height()); // Denominator
+      C.Mult(w,D);
+
+      R = val;
+      for (int i = 0; i<J.Size(); i++)
+      {
+         int ii = J[i];
+         R(ii) = N(ii)/D(ii);
+      }
+
+      Vector verr(val);
+      verr-=R;
+
+      if (verr.Normlinf() <= tol*val.Normlinf()) { break; }
+   }
+}
+
+void ComputePolesAndZeros(const Vector &z, const Vector &f, const Vector &w,
+                          Array<double> & poles, Array<double> & zeros, double &scale)
+{
+   // Initialization
+   poles.SetSize(0);
+   zeros.SetSize(0);
+
+   // Compute the poles
+   int m = w.Size();
+   DenseMatrix B(m+1); B = 0.;
+   DenseMatrix E(m+1); E = 0.;
+   for (int i = 1; i<=m; i++)
+   {
+      B(i,i) = 1.;
+      E(0,i) = w(i-1);
+      E(i,0) = 1.;
+      E(i,i) = z(i-1);
+   }
+
+#ifdef MFEM_USE_LAPACK
+   DenseMatrixGeneralizedEigensystem eig1(E,B);
+   eig1.Eval();
+   Vector & evalues = eig1.EigenvaluesRealPart();
+   for (int i = 0; i<evalues.Size(); i++)
+   {
+      if (IsFinite(evalues(i)))
+      {
+         poles.Append(evalues(i));
+      }
+   }
+#else
+   mfem_error("Compiled without LAPACK");
+#endif
+   // compute the zeros
+   B = 0.;
+   E = 0.;
+   for (int i = 1; i<=m; i++)
+   {
+      B(i,i) = 1.;
+      E(0,i) = w(i-1) * f(i-1);
+      E(i,0) = 1.;
+      E(i,i) = z(i-1);
+   }
+
+#ifdef MFEM_USE_LAPACK
+   DenseMatrixGeneralizedEigensystem eig2(E,B);
+   eig2.Eval();
+   evalues = eig2.EigenvaluesRealPart();
+   for (int i = 0; i<evalues.Size(); i++)
+   {
+      if (IsFinite(evalues(i)))
+      {
+         zeros.Append(evalues(i));
+      }
+   }
+#else
+   mfem_error("Compiled without LAPACK");
+#endif
+
+   scale = w * f / w.Sum();
+}
+
+
+void PartialFractionExpansion(double scale, Array<double> & poles,
+                              Array<double> & zeros, Array<double> & coeffs)
+{
+   int psize = poles.Size();
+   int zsize = zeros.Size();
+   coeffs.SetSize(psize);
+   coeffs = scale;
+
+   for (int i=0; i<psize; i++)
+   {
+      double tmp_numer=1.0;
+      for (int j=0; j<zsize; j++)
+      {
+         tmp_numer *= poles[i]-zeros[j];
+      }
+
+      double tmp_denom=1.0;
+      for (int k=0; k<psize; k++)
+      {
+         if (k != i) { tmp_denom *= poles[i]-poles[k]; }
+      }
+      coeffs[i] *= tmp_numer / tmp_denom;
+   }
+}
+
+
+// x^-a
+void ComputePartialFractionApproximation(double alpha,
```

Comment:
Some guidance on how to select the default parameters is necessary. A reference should be added to the paper describing the approximation.